### PR TITLE
Rename LCU block encodings

### LINKED ISSUE #1240: LCUBlockEncoding / LCUBlockEncodingZeroState names and dangers
This came up in the sync but perhaps there's a better naming convention. Also one danger of LCUBlockEncoding is that the T count is only for Select (maybe this is ok but maybe a better name would prevent confusion)

### PR DESCRIPTION
**Pull request by fdmalone**
Fixes #1240

### CODE REVIEW

File: qualtran/bloqs/block_encoding/lcu_block_encoding.py [hunk 1]
```diff
@@ -369,17 +369,16 @@ def _black_box_lcu_zero_state_block() -> LCUBlockEncodingZeroState:
...
     t = 1
     prepare = PrepareHubbard(x_dim=dim, y_dim=dim, t=t, u=U)
-    black_box_lcu_zero_state_block = LCUBlockEncodingZeroState(
+    black_box_lcu_zero_state_block = LCUBlockEncoding(
         select=BlackBoxSelect(select), prepare=BlackBoxPrepare(prepare)
     )
     return black_box_lcu_zero_state_block
 
 
 _LCU_BLOCK_ENCODING_DOC = BloqDocSpec(
```
Comment (anurudhp) [inline qualtran/bloqs/block_encoding/lcu_block_encoding.py:378]: rename

File: qualtran/bloqs/block_encoding/lcu_block_encoding.py [hunk 2]
```diff
@@ -369,17 +369,16 @@ def _black_box_lcu_zero_state_block() -> LCUBlockEncodingZeroState:
...
-    bloq_cls=LCUBlockEncoding, examples=(_lcu_block, _black_box_lcu_block)
+    bloq_cls=SelectBlockEncoding, examples=(_lcu_block, _black_box_lcu_block)
 )
 
 _LCU_ZERO_STATE_BLOCK_ENCODING_DOC = BloqDocSpec(
```
Comment (anurudhp) [inline qualtran/bloqs/block_encoding/lcu_block_encoding.py:382]: rename

File: qualtran/bloqs/block_encoding/lcu_block_encoding.py [hunk 3]
```diff
@@ -176,7 +176,7 @@ def get_single_qubit_controlled_bloq(self, control_val: int) -> 'LCUBlockEncodin
 
 
 @attrs.frozen
-class LCUBlockEncodingZeroState(BlockEncoding, SpecializedSingleQubitControlledExtension):
+class LCUBlockEncoding(BlockEncoding, SpecializedSingleQubitControlledExtension):
     r"""LCU based block encoding using SELECT and PREPARE oracles.
```
Comment (anurudhp) [inline qualtran/bloqs/block_encoding/lcu_block_encoding.py:180]: add a line explaining how this differs from `SelectBlockEncoding`?

Comment (anurudhp) [reply qualtran/bloqs/block_encoding/lcu_block_encoding.py:180]: Also the docstring below should use signal state 0 instead of G. (lines 197-199)

**Review by anurudhp — APPROVED** [review]
lgtm % doc nits

File: qualtran/bloqs/block_encoding/lcu_block_encoding.py (hunk 1 above)
Comment (fdmalone) [reply qualtran/bloqs/block_encoding/lcu_block_encoding.py]: Done.

File: qualtran/bloqs/block_encoding/lcu_block_encoding.py (hunk 2 above)
Comment (fdmalone) [reply qualtran/bloqs/block_encoding/lcu_block_encoding.py]: Done.

File: qualtran/bloqs/block_encoding/lcu_block_encoding.py (hunk 3 above)
Comment (anurudhp) [reply qualtran/bloqs/block_encoding/lcu_block_encoding.py:180]: Ah I misread the `I_a`, yeah this is fine. But still I feel it is better to just avoid G altogether and just say <0|B[H]|0>, makes it simpler to read. But either is fine.

Comment (fdmalone) [reply qualtran/bloqs/block_encoding/lcu_block_encoding.py:180]: I will defer to you.

**Pull request merged**